src/storage-client/src/controller/statistics.rs

```diff
@@ -0,0 +1,106 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+//! A tokio task (and support machinery) for producing storage statistics
+
+use std::any::Any;
+use std::collections::HashMap;
+use std::sync::{Arc, Mutex};
+use std::time::Duration;
+
+use itertools::Itertools;
+use timely::progress::ChangeBatch;
+use tokio::sync::oneshot;
+
+use mz_ore::cast::CastFrom;
+use mz_repr::{Datum, GlobalId, Row};
+
+use crate::client::SourceStatisticsUpdate;
+use crate::controller::collection_mgmt::CollectionManager;
+
+/// Spawns a task that continually (at an interval) write statistics from storaged's
+/// that are consolidated in shared memory in the controller.
+pub(super) fn spawn_statistics_sender(
+    statistics_collection_id: GlobalId,
+    collection_mgmt: CollectionManager,
+    shared_stats: Arc<Mutex<HashMap<GlobalId, HashMap<usize, SourceStatisticsUpdate>>>>,
+) -> Box<dyn Any + Send + Sync> {
+    // TODO(guswynn): Should this be configurable? Maybe via LaunchDarkly?
+    const STATISTICS_INTERVAL: Duration = Duration::from_secs(30);
+
+    let (shutdown_tx, mut shutdown_rx) = oneshot::channel::<()>();
+
+    mz_ore::task::spawn(|| "statistics_sender", async move {
+        // Keep track of what we think the contents of the output
```

Comment:
nit: this sentence seems to be missing some words and/or the syntax seems off.

Also, though: this is from my original code, so my fault! 😅